[RJA-1067][feature]: UI for variables grouping.

ddurnev opened the pull request:
Required groups not supported yet.

Commit:
[RJA-1067][feature]: UI for variables grouping.
Required groups not supported yet.

bugzmanov commented on ui/src/main/resources/app/modules/createenv.js:
```diff
@@ -212,9 +212,14 @@ function(genesis, backend,  status, variables, gtemplates, validation, Backbone,
     preconditionErrorTemplate: "app/templates/createenv/preconditions_error.html",
 
+    events: {
+      "click .group-radio": function(e) {variables.groupVarSelected(e, this, this.variables);}
```
bad styling.

  on-line function declaration suites for those places where in scala you'd use closures with one statement in it, like map, foreach.
callbacks declaration is not the case.

bugzmanov commented on ui/src/main/resources/app/modules/createenv.js:
```diff
@@ -111,7 +111,7 @@ function(genesis, backend,  status, variables, gtemplates, validation, Backbone,
 
     createEnvironment: function() {
-      if($('#environment-settings').valid()) {
+      if($('#workflow-parameters-form').valid()) {
```
is global $ really neeaded here?

bugzmanov commented on ui/src/main/resources/app/modules/env_details/env_details.js:
```diff
@@ -380,6 +380,11 @@ function (genesis, backend, poller, status, EnvHistory, EnvAccess, variables, gt
+    templateVars: "app/templates/common/variables.html",
+
+    events: {
+      "click .group-radio": function(e) {variables.groupVarSelected(e, this, this.workflow.variables);}
```
bad style

bugzmanov commented on ui/src/main/resources/app/variables.js:
```diff
@@ -143,6 +143,18 @@ function(genesis, status, $, _) {
+       var $currentTarget = view.$(event.currentTarget);
+       var group = $currentTarget.attr('name');
+       var name = $currentTarget.attr('data-var-name');
+       view.$("[id=" + name + "]").show();
```
why not  view.$("#" + name)  ?

bugzmanov commented on ui/src/main/resources/app/variables.js:
```diff
@@ -143,6 +143,18 @@ function(genesis, status, $, _) {
+       var name = $currentTarget.attr('data-var-name');
+       view.$("[id=" + name + "]").show();
+       _.each(variables.filter(function(v){return group == v.group;}), function (x) {
+         if (x.name !== name) {
```
why x.name !== name is not in filter expression?

Commit:
[RJA-1067][feature]: UI for variables grouping.
Fixed checkbox, radio inputs.
Disable non-selected variables in group.
Pre-select & enable variable if it has default value.
